src/components/plugin/Plugin.js

```diff
@@ -23,6 +26,28 @@ const d2Config = {
     ],
 }
 
+const query = {
+    systemSettings: {
+        resource: 'systemSettings',
+        params: {
+            key: SYSTEM_SETTINGS,
+        },
+    },
+}
+
+const providerDataTransformation = ({ systemSettings }) => {
+    return {
+        systemSettings: Object.assign(
+            {},
+            DEFAULT_SYSTEM_SETTINGS,
+            systemSettings,
+            {
+                hiddenPeriods: getHiddenPeriods(systemSettings),
```

Comment:
Is `hiddenPeriods` needed in the plugin?

Reply:
Good point. I don't think so. Also, I'm going to use useCallback here to get rid of the re-rendering in the plugin.